lib/mongoid/contextual/memory.rb

```diff
@@ -164,6 +164,39 @@ def last
         eager_load([documents.last]).first
       end
 
+      # Take the given number of documents from the database.
+      #
+      # @example Take a document.
+      #   context.take
+      #
+      # @param [ Integer ] value The number of documents to take.
+      #
+      # @return [ Document ] The document.
+      def take(value = nil)
```

Comment:
```suggestion
      def take(limit = nil)
```

Reply:
I don't think I can use `limit` as the parameter name since we already have a method called `limit` that's used in this method) and there will be collisions between the names

Reply:
Actually, looks like it works!

Reply:
You can call the method via `self.limit` to disambiguate when not passing it any arguments, same to access an instance attribute (technically also a method) when there is a local variable of the same name.